nameserver/server.go

```diff
@@ -32,7 +20,94 @@ func makeDNSFailResponse(r *dns.Msg) *dns.Msg {
 	return m
 }
 
-func queryHandler(lookups []Lookup) dns.HandlerFunc {
+type DNSServer struct {
+	config *dns.ClientConfig
+	zone   Zone
+	iface  *net.Interface
+	port   int
+	udpSrv *dns.Server
+	tcpSrv *dns.Server
+}
+
+// Creates a new DNS server with a given config
+func NewDNSServerWithConfig(config *dns.ClientConfig, zone Zone, iface *net.Interface, port int) (*DNSServer, error) {
```

Comment:
I don't see the point of the error return. If at some future point we really want to return an error, it's easy enough to change the API.